plotly/plotly/plotly.py

```diff
@@ -1484,6 +1484,16 @@ def get_grid(grid_url, raw=False):
 
     # make request to grab the grid id (fid)
     r_meta = requests.get(meta_get_url, headers=headers)
+
+    # check if request is ok
+    if not r_meta.ok:
+        raise exceptions.PlotlyError(
+            "Whoops, you got a 404 response. {} may be "
+            "pointing to a plot rather than a grid. If it is a grid, make "
+            "sure your grid is 'public' and not 'private' or "
+            "'secret'.".format(grid_url)
+        )
```

Comment:
💀 ! How do you know it's a 404?!? Just do `r_meta.raise_for_status()`. Vague is better than wrong 😸. If you really want a gold ⭐️, you can write a function that takes in a `requests.Response` object and raises with the most pertinent information it can get at.

I.e., this can also be shared with the other error-handling code you're writing right now.